### PR TITLE
Feature/api docs

### PR DESCRIPTION
**Pull request by aussig**
I've added small HTTP API and Web Socket API sections, moved the relevant existing text into the web socket APi section and documented the two HTTP API endpoints.

Also created a separate 'Hosted Service' section to describe the services provided on https://tick.edcd.io/

Can preview the changes here of course: https://github.com/aussig/TickDetector/tree/feature/api-docs#readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,3 +107,67 @@ You *can* instead adjust `tick.html` and `tick.js` to run the WebSocket on
 a separate port, but then you'll also need to adjust both `Dockerfile` and
 `docker-compose.yml` to expose that extra port, along with possibly needing
 to allow that extra port through any firewall you have.
+## API Reference
+
+### Web Socket API
+As part of providing the main web page `tick.js` provides a Web Socket
+which will receive a `'message'` when there is a new tick.
+
+This socket is provided on the standard HTTP/HTTPS port, using whichever URI
```
Comment (Athanasius) [inline README.md:116]: I know this is *my* wording, but technically it has to be using the `ws(s)` protocol.  I seem to recall that trying to fake this in curl (without websocket support) by setting the appropriate extra headers Just Did Not Work.

It's just that the host and port aspects of the URI are indeed the same as the HTTP(s) ones.

Comment (aussig) [reply README.md:116]:
```suggestion
This socket is provided over the `ws(s)` protocol on the standard HTTP/HTTPS port, using the same URI that
```

**Review by aussig — COMMENTED** [review]
I've made a slight amend to the lines you flagged, incorporating mention of the wss protocol.

File: README.md (hunk 1 above)
Comment (Athanasius) [reply README.md:116]: Actually I'll fix this bit up after merge, as there are some other subtleties, depending on what you're using to make the websocket connection.

**Pull request merged**